DataFormats/HGCDigi/bin/HGCSampleTest.cpp

```diff
@@ -0,0 +1,107 @@
+#include "DataFormats/HGCDigi/interface/HGCSample.h"
+#include <iostream>
+#include <cassert>
+#include <string>
+
+// run for instance with:
+//
+//                         time HGCSampleTest  10000000000
+//                         time HGCSampleTest  1000000000 => 14 sec ; 11 sec after restructuring set()
+//                         time HGCSampleTest  10000000000    y   [to require randomisation]
+//
+// for a measureble amount of time taken
+
+int main(int argc, char** argv) {
+  std::cout << "Basic performance tests for HGCSample\n" << std::endl;
+  std::cout << "num parameters entered: " << argc << std::endl;
+
+  // first command line argument is the number of trials
+  unsigned long int repetitions = 100;
+  if (argc > 1)
+    repetitions = std::stoul(argv[1], nullptr, 0);
+  std::cout << "\t + repetitions [int]: " << repetitions << std::endl;
+  // second command line argument (whatever it is) will activate
+  //                  the random choice of values for all inputs
+  bool generateRandomValues = (argc > 2 ? true : false);
+  std::cout << "\t + generateRandomValues [true/false]: " << generateRandomValues << "\n" << std::endl;
+
+  // init static values
+  uint32_t adc = 124;
+  uint32_t gain = 15;
+  uint32_t thrADC = 900;
+  bool thr = adc > thrADC;
+  uint32_t toa = 0;
+  bool mode = false;
+  bool toaFired = true;
+
+  // do the trials: time/performance test and exploit randomisation to check
+  unsigned long int u = 0;
+  for (; u < repetitions; u++) {
+    // randomise all inputs, if chosen at the command line
+    if (generateRandomValues) {
+      adc = rand() % 4096;
```

Comment:
better to use the more modern [C++ random number library](https://en.cppreference.com/w/cpp/numeric/random) rather than `rand()`

Reply:
tks, addressed